server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -71,44 +74,42 @@ public MessageFactory(MessagePreviewGenerator messagePreview, MessageContentExtr
         this.messageContentExtractor = messageContentExtractor;
     }
 
-    public Message fromMessageResult(MessageResult messageResult,
-            List<MessageAttachment> attachments,
-            MailboxId mailboxId,
-            Function<Long, MessageId> uidToMessageId) throws MailboxException {
-        MessageId messageId = uidToMessageId.apply(messageResult.getUid());
+    public Message fromMetaDataWithContent(MetaDataWithContent message) throws MailboxException {
 
-        MessageBuilder parsedMessageResult;
+        org.apache.james.mime4j.dom.Message mimeMessage;
```

Comment:
that's weird to split the method in 3 parts : var declaration (aka C-style), loading of some stuff (+exception handling), builder call.
I'd rather create submethods if needed.